### ISSUE TITLE
AdGuard for Android Disturbs VOIP Apps

### ISSUE THREAD

**Issue by AntiAd** [AdguardTeam/AdguardForAndroid#5263]
### Please answer the following questions for yourself before submitting an issue

- [X] Filters were updated before reproducing an issue
- [X] I checked the [knowledge base](https://adguard.com/kb/) and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

4.3.1

### Environment

- OS version: Android 10 to 14
- Devices: PocoFone F1, Red Note 8 Pro, OnePlus 13

### HTTPS filtering

- [X] yes, I do

### Root access

- [ ] yes, I have it

### Integration with AdGuard VPN

- [ ] yes, I do

### Routing mode

Local VPN

### Ad Blocking

AdGuard Base filter, AdGuard Mobile Ads filter, EasyList

### Privacy

AdGuard Tracking Protection filter, AdGuard URL Tracking filter, EasyPrivacy, Legitimate URL Shortener, Peter Lowe's Blocklist, Fanboy's Anti-Facebook List, Fanboy's Anti-thirdparty Fonts

### Social

AdGuard Social Media filter, Fanboy's Social Blocking List

### Annoyances

AdGuard Annoyances filter, AdGuard Cookie Notices filter, AdGuard Popups filter, AdGuard Mobile App Banners filter, AdGuard Other Annoyances filter, AdGuard Widgets filter, Adblock Warning Removal List, Fanboy's Annoyances, I don't care about cookies, EasyList Cookie List, Dandelion Sprout's Annoyances List

### Security

Online Malicious URL Blocklist, NoCoin Filter List

### Other

Filter unblocking search ads and self-promotion, AdGuard DNS filter, AdGuard Experimental filter

### Which DNS server do you use?

AdGuard DNS

### DNS protocol

DNS-over-HTTPS

### Issue Details

Steps to reproduce:
1. Enable AdGuard
2. Apply the mentioned settings
3. Try to make or receive a call using VOIP apps such as TextNow, 2ndLine, TextFree, etc.

### Expected Behavior

Blocking Ads in VOIP apps such as TextNow, 2ndLine, FreeText, etc., withoutght interrupting their functionality

### Actual Behavior

AdGuard for Android has disturbed VOIP apps such as TextNow, 2ndLine, TextFree, etc., so badly that it literally makes them impossible to use:
Turning AdGuard on causes extremely long delays when:
- Making a call.
- Receiving calls. It lets you receive missed calls, or even if it rings, the call cannot get connected and be answered (ringing, or remains connecting, and hangs out after a long waiting time).

The problem will be solved by disabling AdGuard, but it ends up drowning in a sea of Ads.
Tested on Android 10 to 14. The problem is much more severe in newer versions (i.e., Android 14).

Note: This problem is not observed in iOS.

**Comment by maxikuzmin** [comment]
@AntiAd please try to exclude the app from filtering
You need to go to:
App management -> VOIP -> disable "Route traffic through AdGuard"

Please let me know the result

**Comment by AntiAd** [comment]
@maxikuzmin Thanks for your response, but obviously, it is not a realistic solution for this bug. As explained, the problem will be solved by excluding the apps to be protected by AdGuard or simply by "disabling AdGuard, but **it ends up drowning in a sea of Ads**." I hope for a "_solution_" for this "_bug_" in AdGuard.

**Comment by AntiAd** [comment]
@maxikuzmin Thanks for your response, but obviously, it is not a solution. As explained, "_the problem will be solved by disabling AdGuard_", or excluding the apps to be protected by AdGuard. It is an indicator that the problem arises by AdGuard, but **it ends up drowning in a sea of Ads**." I hope for a "_solution_" for this "_bug_" in AdGuard to have the "_expected behavior_", which is "_blocking Ads in VOIP apps such as TextNow, 2ndLine, FreeText, etc., without interrupting their functionality._"

**Comment by AntiAd** [comment]
@maxikuzmin Thanks for your response, but obviously, it is not a solution. As explained, "_the problem will be solved by disabling AdGuard_", or excluding the apps to be protected by AdGuard. It is an indicator that the problem arises by AdGuard, but "**_it ends up drowning in a sea of Ads_**." I hope for a _solution_ for this AdGuard _bug_ leading to the _Expected Behaviour_, which is "_blocking Ads in VOIP apps such as TextNow, 2ndLine, FreeText, etc., without interrupting their functionality._"

**Comment by Maghsad** [comment]
The same frustrating problem for over one year. Purchased AdGuard just to filter such Ads. It worked at the beginning but led to absolute dissatisfaction since the problem is not solved yet. The problem started with 2ndLine, and after several months expanded to Textnow. 

Considering user complaints on other platforms such as Reddit, some of which go back several years ago, this is a serious AdGuard Bug and is supposed to be considered as P1.

**Comment by maxikuzmin** [comment]
@Maghsad if similar issues are observed with other apps, please open a new issue and provide details of the problem and steps to reproduce it

**Comment by maxikuzmin** [comment]
@AntiAd it would be nice to get your logs.

* Enable debug logging:
Settings -> General -> Advanced -> Logging level -> Debug

* Reproduce the issue, then remember the exact time when it happened

* Collect logs:

Settings -> General -> Advanced -> Export logs and system info

* Sent this file to [devteam@adguard.com](mailto:devteam@adguard.com):
mention the 5263 number in the subject
specify the exact time when the issue occurred

**Comment by maxikuzmin** [comment]
@AntiAd I checked the Talk 360 app, the calls are going well, no problems, the Talk 360 app also uses VOIP

**Comment by Maghsad** [comment]
@AntiAd: As mentioned by @maxikuzmin, I am also not quite sure VOIP is the best term for categorizing this serious AdGuard bug, as anybody may mistakenly refer to apps which are not affected by this serious bug such as WhatsApp and Telegram, etc. Maybe "virtual phone number" or something similar would be a better term.

@maxikuzmin: I faced this problem with all apps mentioned in this bug report:
- Textnow
- 2ndLine
- TextFree

The problem is the same as what is described in this bug report (Extremely long delay in making and answering calls, leading to missing calls). It may be worth mentioning that Textnow and 2ndLine seem so similar that they may use the same technologies (same company or holding?). Based on my experience, 2ndLine seems like the playground of Textnow as the problems first start in 2ndLine and then expand to TextNow. It is almost impossible to make calls with 2ndLine while AdGuard is on (or 2ndLine traffic passing through AdGuard). Please let me know if I should report it in a separate support request.

**Comment by maxikuzmin** [comment]
@Maghsad can I ask you to send debug logs as indicated in the instructions above?
Unfortunately I don't have the ability to check the specified applications.
As far as I know, a similar bug was fixed 3 years ago, but we need to see the logs to fully understand it

**Comment by maxikuzmin** [comment]
@Maghsad @AntiAd any news?

**Comment by maxikuzmin** [comment]
@AntiAd for the moment, I’m closing this issue. You can reopen the issue in the future if the problem persists
